boa/src/value/operations.rs

```diff
@@ -134,12 +134,15 @@ impl Value {
     pub fn rem(&self, other: &Self, ctx: &mut Context) -> Result<Value> {
         Ok(match (self, other) {
             // Fast path:
-            (Self::Integer(x), Self::Integer(y)) => Self::integer(x % *y),
+            (Self::Integer(x), Self::Integer(y)) => Self::rational(f64::from(*x) % f64::from(*y)),
```

Comment:
```suggestion
            (Self::Integer(x), Self::Integer(y)) => {
            	if y == 0 {
            		Value::nan()
            	} else {
            		x % y
            	}
            },
```
Its better to almost always return an integer rather than floating point.

Reply:
Done